homeassistant/components/advantage_air/number.py

```diff
@@ -0,0 +1,54 @@
+"""Number platform for Advantage Air integration."""
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.const import ENTITY_CATEGORY_CONFIG, TIME_MINUTES
+
+from .const import DOMAIN as ADVANTAGE_AIR_DOMAIN
+from .entity import AdvantageAirEntity
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up AdvantageAir toggle platform."""
+
+    instance = hass.data[ADVANTAGE_AIR_DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for ac_key in instance["coordinator"].data["aircons"]:
+        entities.append(AdvantageAirTimeTo(instance, ac_key, "On"))
+        entities.append(AdvantageAirTimeTo(instance, ac_key, "Off"))
+    async_add_entities(entities)
+
+
+class AdvantageAirTimeTo(AdvantageAirEntity, NumberEntity):
+    """Representation of Advantage Air TimeTo number."""
+
+    _attr_unit_of_measurement = TIME_MINUTES
+    _attr_entity_category = ENTITY_CATEGORY_CONFIG
+    _attr_step = 1
+    _attr_min_value = 0
+    _attr_max_value = 720
+
+    def __init__(self, instance: str, ac_key: str, action: str) -> None:
+        """Initialize the Advantage Air timer number."""
+        super().__init__(instance, ac_key)
+        self._time_key = f"countDownTo{action}"
+        self._attr_name = f'{self._ac["name"]} Time To {action}'
+        self._attr_unique_id = (
+            f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-timeto{action}'
+        )
+
+    @property
+    def value(self):
+        """Return the current value."""
+        return self._ac.get(self._time_key)
```

Comment:
My impression is that this is returning time until a specific state is reached? This means a lot of updates to the state machine, and an absolute time is preferred instead. Am i understanding right?

Reply:
An absolute time would have to be inferred, and may not be accurate, as the appliance only provides the minutes remaining on its API. I expect using an inferred  absolute might cause the value to flick back and forward every 30 seconds between two times because no seconds value is understood.

You are correct that the state machine would be updated, but only every minute.

Reply:
Yeah I imagine it would also have to include a setting for only updating when it's a significant change (my impression is there is an API for that)

I believe there was an arch discussion on this topic that forbid using relative times. Am I remembering that right? I couldn't find it quickly.

Reply:
We really try to avoid something that writes state updates due to a running timer.

I'd still convert this to an absolute time unit even if it has to be the ceiling value and displays for a few seconds longer than it actually runs.  This integration is polling so its already expected it won't be in sync with the device for a few seconds

Reply:
@bdraco is the number platform appropriate for absolute time? I dont actually know which platform has the right interfaces for setting absolute time, and I am going to run into issues because the hardware wont support times over 720 minutes in the future.

Reply:
I'm pretty sure that only `sensor` has native support for `datetime` objects

Reply:
Yeah right, so I need to trash this PR and leave the existing implementation which is a sensor and custom service.

Reply:
I think it makes sense to add support for absolute time to the number platform.

That will change the entity model though so it needs an arch discussion https://github.com/home-assistant/architecture/discussions